packages/liveblocks-react-tiptap/src/LiveblocksExtension.ts

```diff
@@ -332,20 +332,30 @@ export const useLiveblocksExtension = (
       ];
     },
     addStorage() {
-      if (!providersMap.has(room.id)) {
+      /* NOTE:
+      the purpose of the providersMap is to keep the same provider alive across renders
+      re-instantiating the provider when the room hasn't changed can cause the yjs doc state to be out of sync
+      with liveblocks yjs state. In this instance, we can just check if the room has changed, because the instance
+      from useRoom will be shallowly equal to the previous instance.
+      */
+      let provider = providersMap.get(room.id);
+      if (provider && provider.room !== room) {
+        // The room has changed, destroy the old provider, so a new one can be made
+        provider.destroy();
+        providersMap.delete(room.id);
+      }
```

Comment:
2/2

…it means this solution can likely be implemented in a simpler way, by not keying these in a Map from room _ID_ to providers, but instead by using a WeakMap of Room _instances_ to providers.

This has been our go-to trick to extend stuff on the Room in other packages, without the Room having to know about it. And if the Room instance goes out of existance, the weakmap will also garbage-collect these provider instances.